assets/js/blocks/cart-checkout/checkout-i2/inner-blocks/checkout-terms-block/frontend.tsx

```diff
@@ -19,23 +18,25 @@ const FrontendBlock = ( {
 	text,
 	checkbox,
 	instanceId,
+	validation,
 }: {
 	text: string;
 	checkbox: boolean;
 	instanceId: string;
+	validation: ( validationErrorId: string ) => ValidationData;
 } ): JSX.Element => {
 	const [ checked, setChecked ] = useState( false );
-
 	// @todo Checkout i2 - Pass validation context to Inner Blocks to avoid exporting in a public package.
 	const { isDisabled } = useCheckoutSubmit();
```

Comment:
Is this todo incorrect here? I'm assuming you mean passing the `checkoutSubmit` context to Inner Blocks to avoid exporting?

Reply:
Yeah, it seems to be left over, fixed in 3e10485

Reply:
I see the change, however this block is still importing `useCheckoutSubmit` which is only available because the block is built as a part of the Woo Blocks repo. Is `useCheckoutSubmit` exposed to extension developers as well? If not, the `checkout-terms-block` would be a good reference for extension developers to learn how to create this type of block so we might want to consider exposing the values of `useCheckoutSubmit` to extensions?

This might already be considered as a followup, or elsewhere. If so feel free to ignore my comment :)

Reply:
> Is useCheckoutSubmit exposed to extension developers as well?
No.

> the checkout-terms-block would be a good reference for extension developers to learn how to create this type of block

Yes, but also no. We have [Newsletter Block](https://github.com/opr/newsletter-test) that is optimized to do that. 

> so we might want to consider exposing the values of useCheckoutSubmit to extensions?

We previously discussed this, `useCheckoutSubmit` was exposed to be used in Newsletter Block (very similar code given both are checkboxes) but I decided we withheld from that.

[My reasoning was](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/4607#discussion_r698450400) that `useCheckoutSubmit` is an internal hook that use to render the submit button, it didn’t make sense to expose it for this reason, a better/dedicated hook for checkout states should be exposes if this proves to be needed.